src/engine/GroupBy.h

```diff
@@ -156,6 +156,42 @@ class GroupBy : public Operation {
   // `?z`.
   bool computeGroupByForJoinWithFullScan(IdTable* result);
 
+  // TODO: Update documentation
+  template <size_t OUT_WIDTH>
+  bool computeGroupByForSortedSubtree(IdTable* result,
+                                      std::vector<Aggregate> aggregates,
+                                      const IdTable& subresult,
+                                      size_t columnIndex,
+                                      LocalVocab& localVocab);
+
+  struct ExplicitlySortedData {
+    size_t subtreeColumnIndex_;
+  };
+
+  // Will be the implementation of an interface containing an
+  // "incrementor" function, a result function, a child expression
+  // or expression is passed to "increment" function, along with evalContext?
+  struct AverageAggregationData {
+    double _sum;
+    int _count;
+    void increment(double intermediate) {
+      _sum += intermediate;
+      _count++;
+    };
+    double calculateResult() const { return _sum / _count; }
```

Comment:
Initialize the default values to zero. Then you get to use the HashMap trick I mentioned in the other file.
Also: The `_count` should be a bigger type (use `int64_t`, that's what the expression module also uses), otherwise it will break at 2 billion values.